--- openmpi-3.1.4/ompi/mca/fs/lustre/fs_lustre.c	2019-04-15 12:49:43.000000000 -0700
+++ patch-files/fs_lustre.c	2021-08-31 17:59:14.412830014 -0700
@@ -23,6 +23,9 @@
  * modules just to query their version and parameters
  */
 
+#ifdef HAVE_SYS_MOUNT_H
+#include <sys/mount.h>
+#endif
 #include "ompi_config.h"
 #include "mpi.h"
 #include "ompi/mca/fs/fs.h"
@@ -35,9 +38,6 @@
 #ifdef HAVE_SYS_PARAM_H
 #include <sys/param.h>
 #endif
-#ifdef HAVE_SYS_MOUNT_H
-#include <sys/mount.h>
-#endif
 #ifdef HAVE_SYS_STAT_H
 #include <sys/stat.h>
 #endif
