--- openmpi-3.1.4/ompi/mca/fs/lustre/fs_lustre.c	2019-04-15 12:49:43.000000000 -0700
+++ patch-files/fs_lustre.c	2019-08-12 22:22:20.592966239 -0700
@@ -88,7 +88,7 @@
     tmp = strchr (fh->f_filename, ':');
     if (!tmp) {
         if (OMPIO_ROOT == fh->f_rank) {
-            fh->f_fstype = mca_fs_base_get_fstype ( fh->f_filename );
+            fh->f_fstype = mca_fs_base_get_fstype ( (char *) fh->f_filename );
         }
 	fh->f_comm->c_coll->coll_bcast (&(fh->f_fstype),
 				       1,
