--- openmpi-1.8.4/ompi/mpi/fortran/base/gen-mpi-sizeof.pl	2014-10-31 17:29:50.000000000 -0700
+++ patch-files/gen-mpi-sizeof.pl	2015-05-02 17:33:14.043388965 -0700
@@ -140,7 +140,7 @@
 for my $size (qw/8 16 32 64/) {
     queue_sub("integer(int${size})", "int${size}", "int${size}");
 }
-for my $size (qw/32 64 128/) {
+for my $size (qw/32 64/) {
     queue_sub("real(real${size})", "real${size}", "real${size}");
     queue_sub("complex(real${size})", "complex${size}", "real${size}");
 }
