--- mvapich2-2.3/contrib/hwloc/config/hwloc.m4	2018-07-24 07:30:00.000000000 -0700
+++ patch-files/hwloc.m4	2018-08-08 16:15:11.742583581 -0700
@@ -865,12 +865,7 @@
 #error CUDART_VERSION too old
 #endif]], [[int i = 3;]])],
          [AC_MSG_RESULT(yes)
-          AC_CHECK_LIB([cudart], [cudaGetDeviceProperties], [
-            HWLOC_CUDA_LIBS="-lcudart"
-            AC_SUBST(HWLOC_CUDA_LIBS)
-            hwloc_have_cudart=yes
-            AC_DEFINE([HWLOC_HAVE_CUDART], [1], [Define to 1 if you have the `cudart' SDK.])
-          ])
+          hwloc_have_cudart=yes
         ])
       ])
       AC_LANG_POP([C++])
